projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/vdk-jupyterlab-extension/handlers.py

```diff
@@ -54,15 +55,15 @@ def post(self):
 class CreateJobHandler(APIHandler):
     @tornado.web.authenticated
     def post(self):
-        input_data = self.get_json_body()
+        job_data = DictObj(self.get_json_body())
         try:
-            local = True if input_data["isLocal"] else False
-            cloud = True if input_data["isCloud"] else False
+            local = True if job_data.local else False
+            cloud = True if job_data.cloud else False
             status = VdkUI.create_job(
-                input_data["jobName"],
-                input_data["jobTeam"],
-                input_data["restApiUrl"],
-                input_data["jobPath"],
+                job_data.jobName,
```

Comment:
Consider refactoring the VdkUI methods to take just the new job object as an argument. You can access the separate fields inside the methods, instead of passing them here.
```
VdkUI.create_job(job_data)
```

Reply:
I'm not too familiar with Jupyter internals, so take this suggestion as optional.

In fact, if you add the job type to the data model, you could do something like below in every handler.

```
VdkUI.run(job_data)
```

and then inside the run method, you can decide if you want to call create_job, download_job, etc. based on the job type.

Reply:
> Consider refactoring the VdkUI methods to take just the new job object as an argument. You can access the separate fields inside the methods, instead of passing them here.
> 
> ```
> VdkUI.create_job(job_data)
> ```

I really like this idea. It would definitely improve the quality of the handlers.
Tho for the second comment I think having the operations separated is better for now.